internal/bgp/bgp.go

```diff
@@ -168,7 +168,15 @@ func (s *Session) connect() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	deadline, _ := ctx.Deadline()
-	conn, err := dialMD5(ctx, s.addr, s.password)
+
+	laddr := "[::]:0"
+	if (s.routerID.To4() != nil) {
+		laddr = fmt.Sprintf("%s:0", s.routerID.String())
+	} else if (s.routerID.To16() != nil) {
+		laddr = fmt.Sprintf("[%s]:0", s.routerID.String())
```

Comment:
The BGP spec doesn't require the router ID to match any local interfaces. What happens if I specify a router ID of 1.2.3.4, and that address doesn't exist locally? I think there needs to be another check here, to see if the router ID is actually configured on a local interface (`net.Interfaces()` lets you find that out).